File: app/src/main/java/zo/den/testtask4/data/network/pojo/Channel.java
package zo.den.testtask4.data.network.pojo;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.ArrayList;
@Root(name = "channel")
public class Channel {
    @Element
    private String title;
    @Element
    private String description;
    @Element
    private String link;

    @ElementList(entry = "item", inline = true)
    private ArrayList<Item> item;

    public String getTitle ()
    {
        return title;
    }

    public void setTitle (String title)
    {
        this.title = title;
    }

    public String getDescription ()
    {
        return description;
    }

    public void setDescription (String description)
    {
        this.description = description;
    }

    public String getLink ()
    {
        return link;
    }

    public void setLink (String link)
    {
        this.link = link;
    }

    public ArrayList<Item> getItem ()
    {
        return item;
    }

    public void setItem (ArrayList<Item>item)
    {
        this.item = item;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [title = "+title+", description = "+description+", link = "+link+", item = "+item+"]";
    }
}

File: app/src/main/java/zo/den/testtask4/data/network/pojo/Channel.kt
package zo.den.testtask4.data.network.pojo


import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "channel", strict = false)
class Channel {
    @set:ElementList(entry = "link", inline = true, required = false)
    @get:ElementList(entry = "link", inline = true, required = false)
    var links: List<Link>? = null

    @set:ElementList(name = "item", required = true, inline = true)
    @get:ElementList(name = "item", required = true, inline = true)
    var itemList: List<Item>? = null


    @set:Element(name = "title", required = false)
    @get:Element(name = "title", required = false)
    var title: String? = null
    @set:Element(name = "language", required = false)
    @get:Element(name = "language", required = false)
    var language: String? = null

    @set:Element(name = "ttl", required = false)
    @get:Element(name = "ttl", required = false)
    var ttl: Int = 0

    @set:Element(name = "pubDate", required = false)
    @get:Element(name = "pubDate", required = false)
    var pubDate: String? = null

    override fun toString(): String {
        return "Channel{" +
                "links=" + links +
                ", itemList=" + itemList +
                ", title='" + title + '\''.toString() +
                ", language='" + language + '\''.toString() +
                ", ttl=" + ttl +
                ", pubDate='" + pubDate + '\''.toString() +
                '}'.toString()
    }
}
